### ISSUE TITLE
Authentication

### ISSUE THREAD

**Issue by wassertim** [SadatAnwar/lunch-mator#20]
We should think about securing our app and our REST API

- [ ] login screen for the app
- [ ] some token authentication for the REST API

**Comment by SadatAnwar** [comment]
token authentication? 
Okay now I know I am stupid! 
@wassertim go ahead! You are the boss!! But I was thinking of implementing some fancy password encryption algo it would be fun for me :)

**Comment by wassertim** [comment]
Yes token ... if we have some other clients (like iOS app) it should authenticate and get a token which it can use for accessing the REST API.

Yes we should definitely save encrypted passwords.

**Comment by SadatAnwar** [comment]
I think for now we can Put the token authentication into a separate issue, that way it wont block this one

Also, I think what is more important is a signup page, with some company/email address verification 
as in employees from organization are limited to only that.
Maybe this needs to be done on an architectural level
